docs/developer/advanced/sharing-saved-objects.asciidoc

```diff
@@ -0,0 +1,437 @@
+the 8.0 release to minimize confusion and disruption for the end-user experience.
+
+We realize that the conversion process and all that it entails can be a not-insignificant amount of work for some Kibana teams to prepare
+for by the 8.0 release. As long as an object is made share-capable, that ensures that its ID will be globally unique, so it will be trivial
+to make that object shareable later on when the time is right.
+
+A developer can easily flip a switch to make a share-capable object into a shareable one, since these are both serialized the same way.
+However, we envision that each consumer will need to enact their own plan and make additional UI changes when making an object shareable.
+For example, some users may not have access to the Saved Objects Management page, but we still want those users to be able to see what
+space(s) their objects exist them and share them to other spaces. Each application should add the appropriate UI controls to handle this.
+
+
+[[sharing-saved-objects-faq-changing-object-ids]]
+==== 2. Why do object IDs need to be changed?
+
+This is because of how isolated objects are serialized to raw Elasticsearch documents. Each raw document ID today contains its space ID
+(_namespace_) as a prefix. When objects are copied or imported to other spaces, they keep the same object ID, they just have a different
+prefix when they are serialized to Elasticsearch. This has resulted in a situation where many Kibana installations have saved objects in
+different spaces with the same object ID:
+
+image::images/sharing-saved-objects-faq-changing-object-ids-1.png["Sharing Saved Objects object ID diagram (before conversion)"]
+
+Once an object is converted, we need to remove this prefix. Because of limitations with our migration process, we cannot actively check if
+this would result in a conflict. Therefore, we decided to pre-emptively regenerate the object ID for every object in a non-Default space to
+ensure that every object ID becomes globally unique:
+
+image::images/sharing-saved-objects-faq-changing-object-ids-2.png["Sharing Saved Objects object ID diagram (after conversion)"]
+
+[[sharing-saved-objects-faq-multiple-deep-link-objects]]
+==== 3. What if one page has deep links to multiple objects?
+
+As mentioned in <<sharing-saved-objects-q2>>, some URLs may contain multiple object IDs, effectively deep linking to multiple objects.
+These should be handled on a case-by-case basis at the plugin owner's discretion. A good rule of thumb is:
+
+* The "primary" object on the page should always handle the three `resolve()` outcomes as described in <<sharing-saved-objects-step-3>>.
+* Any "secondary" objects on the page should usually handle the first two outcomes the same way, but may vary its behavior for the
+`'conflict'` outcome to ignore it throw an error instead. In the case of an error, an actionable message should be shown to the user.
```

Comment:
__question:__ `may vary its behavior for the 'conflict' outcome to ignore it throw an error instead`. Did you mean `may vary its behavior for the 'conflict' outcome, from ignoring it to throwing an error instead` or something like this?

Reply:
I meant to write "may vary its behavior for the 'conflict' outcome to ignore it _and_ throw an error instead".

I think this can be reworded though. I'll change it and you can take another look.

Reply:
Updated version looks great, thanks!